trl/models/modeling_base.py

```diff
@@ -258,6 +267,14 @@ class and the arguments that are specific to trl models. The kwargs
         if is_resuming_training:
             model.post_init(state_dict=state_dict)
 
+        if not is_peft_model and reward_modeling_adapter_id is not None:
```

Comment:
Do we also need a check that the base model is the same architecture that the reward model adapter (ie check that `llama-7b` is matched to `ybelkada/llama-7b-dummy-rm`